src/main/java/de/varoplugin/cfw/version/OneSevenVersionAdapter.java

```diff
@@ -77,7 +75,53 @@ protected void init() throws IllegalArgumentException, NoSuchMethodException, Se
         this.initRespawn();
         this.initNetworkManager();
         this.initLocale();
-        this.initXp();
+    }
+
+    protected void initXp(Player player) {
+        this.searchXpField(player, VersionUtils.getNmsClass() + ".EntityHuman", VersionUtils.getNmsClass() + ".EntityExperienceOrb");
+    }
+
+    protected void searchXpField(Player player, String humanClassName, String experienceOrbClassName) {
+        try {
+            Class<?> entityHumanClass = Class.forName(humanClassName);
+            Map<Field, Integer> values = new HashMap<>();
+            Object craftPlayer = this.getHandle(player);
+
+            // Get values of all int fields of player
+            for (Field field : entityHumanClass.getDeclaredFields()) {
+                if (field.getType() == int.class) {
+                    field.setAccessible(true);
+                    values.put(field, field.getInt(craftPlayer));
+                }
+            }
+
+
+            // Invoke method which sets the value of xp cooldown to 2
+            Location randomLoc = new Location(player.getWorld(), 0, 0, 0);
+            ExperienceOrb orb = (ExperienceOrb) player.getWorld().spawnEntity(randomLoc, EntityType.EXPERIENCE_ORB);
+            orb.setExperience(0);
+            Class<?> entityExperienceOrbClass = Class.forName(experienceOrbClassName);
+            for (Method method : entityExperienceOrbClass.getDeclaredMethods()) {
+                if (method.getParameterTypes().length == 1 && method.getParameterTypes()[0] == entityHumanClass) {
```

Comment:
`if (method.getParameterTypes().length == 1 && method.getParameterTypes()[0] == entityHumanClass)` looks like it could very easily return the wrong method

Reply:
Yes, this is the part of code which is most likely to be buggy, but still better than the old method since there is only one method with this exact signature parameters.